gulpfile.js

```diff
@@ -22,6 +22,224 @@ const rollup = require('rollup');
+    version: '2.1.1',
+  },
+  querystring: {
+    name: 'querystring-es3',
+    version: '0.2.1',
+  },
+  stream: {
+    name: 'stream-browserify',
+    version: '2.0.2',
+  },
+  string_decoder: {
+    name: 'string_decoder',
+    version: '1.2.0',
+  },
+  timers: {
+    name: 'timers-browserify',
+    version: '2.0.10',
+  },
+  tty: {
+    name: 'tty-browserify',
+    version: '0.0.1',
+  },
+  url: {
+    name: 'url',
+    version: '0.11.0',
+  },
+  util: {
+    name: 'util',
+    version: '0.11.1',
+  },
+  vm: {
+    name: 'vm-browserify',
+    version: '1.1.0',
+  },
+  zlib: {
+    name: 'browserify-zlib',
+    version: '0.2.0',
+  },
+};
+
+const modulesToBrowserify = new Set(Object.keys(browserifyModules));
+
+const transformRxjsImportText = (importName) =>
+  importName === 'EMPTY'
+    ? 'rxjs/internal/observable/empty'
+    : rxjsTypes.has(importName)
+    ? 'rxjs/internal/types'
+    : importName[0].toLowerCase() === importName[0]
+    ? `rxjs/internal/observable/${importName}`
+    : `rxjs/internal/${importName}`;
+
+const transformRxjsOperatorsImportText = (importName) => `rxjs/internal/operators/${importName}`;
+
+const updateImportStatements = (callback, statement) =>
+  statement.importClause.namedBindings.elements.map((elem) =>
+    typescript.createImportDeclaration(
+      statement.decorators,
+      statement.modifiers,
+      typescript.createImportClause(
+        undefined,
+        typescript.createNamedImports([typescript.createImportSpecifier(elem.propertyName, elem.name)]),
+      ),
+      typescript.createStringLiteral(
+        callback(elem.propertyName === undefined ? elem.name.getText() : elem.propertyName.getText()),
+      ),
+    ),
+  );
+
+const RXJS_IMPORT = 'rxjs';
+const RXJS_OPERATORS_IMPORT = 'rxjs/operators';
+
+const transformImports = (statements) =>
+  _.flattenDeep(
+    statements.map((statement) => {
+      if (typescript.isImportDeclaration(statement)) {
+        if (statement.moduleSpecifier.text === RXJS_IMPORT) {
+          return updateImportStatements(transformRxjsImportText, statement);
+        }
+        if (statement.moduleSpecifier.text === RXJS_OPERATORS_IMPORT) {
+          return updateImportStatements(transformRxjsOperatorsImportText, statement);
+        }
+      }
+
+      return statement;
+    }),
+  );
+
+const createCustomTransform = (context) => {
+  return (node) => {
+    const visitor = (node) => {
+      if (typescript.isSourceFile(node)) {
+        const statements = transformImports(node.statements);
+
+        return typescript.updateSourceFileNode(
+          node,
+          statements,
+          node.isDeclarationFile,
+          node.referencedFiles,
+          node.typeReferences,
+          node.hasNoDefaultLib,
+          node.libReferences,
+        );
+      }
+
+      return node;
+    };
+
+    typescript.visitEachChild(node, visitor, context);
+
+    return typescript.visitNode(node, visitor);
+  };
+};
+
+let packageBrowserifyMap = {};
```

Comment:
The gulp steps run in parallel, so you need to make sure the transform package.json always runs after the typescript transforms. Also need to verify that everything runs in the same process. Did you check that the output has the expected dependencies?

Reply:
Ah, I see that you did and it does run afterwards now. Cool.